service/src/main/kotlin/fi/espoo/evaka/assistanceneed/preschooldecision/AssistanceNeedPreschoolDecisionQueries.kt

```diff
@@ -372,6 +349,25 @@ fun Database.Transaction.decideAssistanceNeedPreschoolDecision(
         .updateExactlyOne()
 }
 
+fun Database.Transaction.endActiveAssistanceNeedPreschoolDecisions(
+    excludingId: AssistanceNeedPreschoolDecisionId,
+    endDate: LocalDate,
+    childId: ChildId
+) =
+    createUpdate {
+            sql(
+                """
+UPDATE assistance_need_preschool_decision
+SET valid_to = ${bind(endDate)}
+WHERE id <> ${bind(excludingId)}
+  AND (valid_to IS NULL OR valid_to > ${bind(endDate)})
+  AND child_id = ${bind(childId)}
+  AND status = 'ACCEPTED'
+"""
```

Comment:
Arguably not a very realistic/common case, but currently this fails if you insert a new decision before an old one, as it results in an inverted range. 
Maybe needs `AND valid_from <= ${bind(endDate)}` or some range operator.

Reply:
Fixed